### DISCUSSION TITLE
Aqara H1 WS-EUK04 firmware date 11-10-2021 malfunctioning

### DISCUSSION THREAD

**Discussion by Witek1974** [Koenkk/zigbee2mqtt#16573] [Devices]
I have several Aqara H1 switches, they all work, except that they have a firmware version with the date 01-29-2021. New purchase, date 11-10-2021 and there is a problem. The temperature and energy parameters have a null value. Physical buttons do not change the state of the device.
Do you know a solution for this version to work? Or a way to downgrade the firmware?

Zigbee2MQTT ver. 1.30.1, CC2652P coordinator zStack3x0 ver. 20210211
![image](https://user-images.githubusercontent.com/124715565/217328351-87ba52f6-8a09-4331-bfbe-d36f2b68a959.png)

**Comment by SebastianOsinski** [comment]
I have the same problem with two switches, exact same FW version. Worth to add that WS-EUK03 (single rocker) with the same FW version (at least by the build date) does not have the same problem - both temp and energy are reported correctly and physical button changes the state.

I just noticed this yesterday and thought it might be related to recent update of Zigbee2MQTT. However, I rolled back the whole Home Assistant VM to state from week ago with Z2M 1.29.2 and it looks like the problem was there already.

**Reply by SebastianOsinski** [reply]
I just noticed the same problem with WS-EUK02 with 01-12-2021 FW

**Comment by SebastianOsinski** [comment]
@Witek1974 I managed to fix the issue with all malfunctioning WS-EUK01/02/03/04 I had.

Inspired by [this thread](https://community.home-assistant.io/t/aqara-plug-sp-euc01-lumi-plug-maeu01-not-updating/393110/11) I decided to update my zStack3x0 to [newest FW](https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin). I used [this script](https://github.com/JelmerT/cc2538-bsl) to update the FW. After that I removed all malfunctioning switches from Z2M and repaired them. Now all of them work as they're supposed to - including power consumption and updating state on physical button press

**Reply by Witek1974** [reply]
Great! I will to do the same. Thanks!

**Reply by nevadascout** [reply]
I installed the new firmware earlier but it still didn't seem to work. But once I removed and re-paired the switch it worked perfectly.

Thanks for the tip!

**Comment by KMTsvetanov** [comment]
I'm using the Zigbee 3.0 USB Dongle Plus ZBDONGLE-E (the 5cm in length) ,
If you are like me this is the way:
https://www.youtube.com/watch?v=BU9ha0Uaduc

**Comment by gregbond007** [comment]
I have come across the exact problem described in this post. I have an old WS-EUK04 that is working perfectly on firmware 01-29-2021, and recently received 4x new WS-EUK04s running firmware 11-10-2021 that do not report temperature, power, or allow me to decouple the relays.

I can see a post mentioning upgrading the firmware of the coordinator; however, I use a ConBee2/RaspBee2 running firmware 26580700.

Is there any fix for my situation, such as downgrading the firmware of the WS-EUK04?

**Reply by gregbond007** [reply]
I have updated the firmware on the ConBee2 coordinator to 26780700 following instructions from this manual:
https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Update-deCONZ-manually

This has brought back full functionality, less the ability to couple/decouple the relays. This, however, appears to be a separate issue that has been fixed as per this thread:
https://github.com/Koenkk/zigbee2mqtt/issues/19960

**Comment by rjaros87** [comment]
I've got the same problem. Adapter Skyconnect EZSP v12 version 7.3.2.0 build 212; Z2M: 1.35.3. Switch WS-EUK04 date version: 11-10-2021. Anyone have some solution?

**Comment by casparom** [comment]
Same issue. Bump for solution.

**Comment by Vincentiano** [comment]
Just a headsup for Skyconnect users who still have issues:
The problems are gone for me using the latest zigbee firmware for SkyConnect: 
- Get the latest firmware here https://github.com/darkxst/silabs-firmware-builder/tree/4.4.3/firmware_builds/skyconnect, 
currently this is:  "ncp-uart-hw-v7.4.3.0-skyconnect-115200.gbl"
- Update your skyconnect with a different pc/ laptop using the web flasher 
(https://skyconnect.home-assistant.io/firmware-update/, and select custom firmware)
- Wait till the update is complete
- Plug your SkyConnect back into your original device, and restart your zigbee software
- Force remove your Aqara H1 Switch from Zigbee2MQTT
- Reset the Aqara H1 Switch and rejoin it to Zigbee2MQTT

For Zigbee2MQTT you can also enable the new ember driver, which has been way more responsive for me:
- Go to Settings -> Serial
- For Adapter select: "ember"
- Restart Zigbee2MQTT

**Comment by pjt-15e** [comment]
I've got a different problem with my H1 WS-EUK04 dual switch with firmware 11/10/21.  The Zigbee network coordinator is a Sonoff ZBDongle-P.  There are a currently a couple of Sonoff SNZB-02 thermometers, a SNZB-03 PIR, three SNZB-01 buttons and three SNZB-01P buttons.  The issue is seems to be that the H1 seems to be working in a partial "distributed network" mode and is responding to button presses by toggling one of the switches, when it should be exclusively be controlled by the coordinator and Home Assistant.

Distributed networking is covered in part 8 of https://www.aqara.com/eu/product/wireless-remote-switch-h1/faq/ - "To put it simply, the child devices do not require a hub, and the controller and actuator can be networked in pairs."

I have confirmed that the H1 is in this partial "distributed network" mode as it is possible to stop Home Assistant, the Zigbee2MQTT service, and to remove the coordinator and then still find that the H1 switch toggles when one of the Sonoff SNZB-01P buttons is single pressed!

When first noticed the problem I was using ZHA, Home Assistant's own Zigbee add-in, however I changed-over to Zigbee2MQTT to get better access to logs etc.  Changing from ZHA to Zigbee2MQTT made no difference.  I also updated the firmware on my Sonoff ZBDongle-P (which was a bit of a hassle) without any improvement.

A factory reset (see 5 of the FAQ link above) helps for a while, though is a hassle as the H1 has to re-introduced to the Zigbee network and the relevant Home Assistant routines have to be verified that they still work.

So if anyone has any advise on how to prevent the H1 from adopting the "distributed network" mode I'd appreciate it, or even suggestions on what I'm doing wrong.

Thanks.
